mis_builder_budget/models/mis_budget_by_account_item.py

```diff
@@ -60,6 +60,8 @@ def _compute_balance(self):
 
     def _prepare_overlap_domain(self):
         """Prepare a domain to check for overlapping budget items."""
+        if self.budget_id.allow_items_overlap:
+            return [("id", "=", False)]
```

Comment:
Is it not better to not call `_prepare_overlap_domain` in `_check_dates` when `allow_items_overlap` is set ?